Fix the SMTP auth calculations - use the correct order of arguments.
Only applied if requested at port build time.

--- a/vchkpw.c
+++ b/vchkpw.c
@@ -441,7 +441,8 @@
     /* Check CRAM-MD5 auth */
     if(ConnType == SMTP_CONN) {
       /* printf("vchkpw: smtp auth\n"); */
-      cramaccepted = authcram(ThePass,TheChallenge,vpw->pw_clear_passwd);
+      /* cramaccepted = authcram(ThePass,TheChallenge,vpw->pw_clear_passwd); */
+      cramaccepted = authcram(TheChallenge,ThePass,vpw->pw_clear_passwd);
       if(cramaccepted == 0) strcpy(AuthType, "CRAM-MD5");
     }
 
